examples/send_transfer.py

```diff
@@ -2,45 +2,115 @@
+from iota import (
+  __version__,
+  Address,
+  Iota,
+  ProposedTransaction,
+  Tag,
+  TryteString,
 )
+from six import text_type
 
-# Create the API instance.
-api = Iota(
-    # URI of a locally running node.
-    'http://localhost:14265/',
 
-    # Seed used for cryptographic functions.
-    seed=SEED1,
-)
+def main(address, depth, message, seed, tag, uri, value):
+    # Create the API instance.
+    api = Iota(uri, seed)
+
+    # For more information, see :py:meth:`Iota.send_transfer`.
+    api.send_transfer(
+        depth=depth,
+        # One or more :py:class:`ProposedTransaction` objects to add to the
+        # bundle.
+        transfers=[
+            ProposedTransaction(
+                # Recipient of the transfer.
+                address=Address(address),
+
+                # Amount of IOTA to transfer.
+                # By default this is a zero value transfer.
+                value=value,
 
-# For more information, see :py:meth:`Iota.send_transfer`.
-api.send_transfer(
-    depth=100,
-
-    # One or more :py:class:`ProposedTransaction` objects to add to the
-    # bundle.
-    transfers=[
-        ProposedTransaction(
-            # Recipient of the transfer.
-            address=Address(
-                ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_2,
+                # Optional tag to attach to the transfer.
+                tag=Tag(tag),
+
+                # Optional message to include with the transfer.
+                message=TryteString.from_string(message),
             ),
+        ],
+    )
 
-            # Amount of IOTA to transfer.
-            # This value may be zero.
-            value=1,
+if __name__ == '__main__':
+    parser = ArgumentParser(
+        description=__doc__,
+        epilog='PyOTA v{version}'.format(version=__version__),
+    )
 
-            # Optional tag to attach to the transfer.
-            tag=Tag(b'EXAMPLE'),
+    parser.add_argument(
+        '--address',
+        type=text_type,
+        default=b'RECEIVINGWALLETADDRESSGOESHERE9WITHCHECKSUMANDSECURITYLEVELB999999999999999999999999999999',
+        help=
+        'Receiving address'
+        '(defaults to RECEIVINGWALLETADDRESSGOESHERE9WITHCHECKSUMANDSECURITYLEVELB999999999999999999999999999999).',
+    )
+
+    parser.add_argument(
+        '--depth',
+        type=int,
+        default=3,
+        help=
+        'Depth at which to attach the bundle.'
+        '(defaults to 3).',
+    )
+
+    parser.add_argument(
+        '--message',
+        type=text_type,
+        default='Hello World!',
+        help=
+        'Transfer message.'
+        '(defaults to Hello World!).',
+    )
+
+    parser.add_argument(
+        '--seed',
+        type=text_type,
+        default=b'THESEEDOFTHEWALLETSENDINGGOESHERE99999999999999999999999999999999999999999999999',
+        help=
+        'Users IOTA seed.'
+        '(defaults to THESEEDOFTHEWALLETSENDINGGOESHERE99999999999999999999999999999999999999999999999).',
+    )
```

Comment:
❔ I'm a tad bit uncomfortable having an interface for users to enter their seed in cleartext.  Could we change this to use the functions from https://github.com/iotaledger/iota.lib.py/blob/develop/examples/address_generator.py#L46-L80 instead?

Reply:
Travelling at the moment, but will implement this as soon as I can.

Reply:
Cheers mate!